llvm/lib/Transforms/Coroutines/Coroutines.cpp

```diff
@@ -123,6 +123,17 @@ bool coro::declaresIntrinsics(const Module &M,
   return false;
 }
 
+// Verifies if a module has any intrinsics.
+bool coro::declaresIntrinsics(const Module &M,
+                              const DenseSet<Intrinsic::ID> &Identifiers) {
+  for (const Function &F : M.functions()) {
+    if (Identifiers.contains(F.getIntrinsicID()))
```

Comment:
Can you just make this a switch over the intrinsic ID instead of this weird intermediate set

Reply:
I don't know how the code was originally put together, since it seems to be a subset of coroutine functions that are used here.

Reply:
right so switch over that set?

Reply:
I thought that's what this function does, takes  a subset and checks the module for any matching intrinsic IDs.

Reply:
But it's doing it with this intermediate set. As in, this is an overly fancy, heavy API. I mean literally loop over function declarations with a switch in it, no DenseSet